lib/helpers/request.js

```diff
@@ -0,0 +1,70 @@
+const Base = require('./base');
+const axios = require('axios');
+
+module.exports = class Request extends Base {
+  constructor(base, credentials) {
+    super();
+    // Every instance of Hydrant has a base path and base credentials.
+    this.base = base;
+    this.credentials = credentials;
+  }
+  /**
+   * Issue a generic XMLHttpRequest.
+   * @param {string} method
+   *  The HTTP method to be used in the request.
+   * @param {string} url
+   *  The URL against which to issue the request.
+   * @param {string} XCSRFToken
+   *  An X-CSRF-Token from Drupals REST API.
+   * @param {object} additionalHeaders
+   *  An object containing additional request header key-value pairs.
+   * @param {object} body
+   *  An object containing the request body to be sent.
+   * @returns {Promise}
+   *  A Promise that when fulfilled returns a response from the request.
+   */
+  issueRequest(method, url, XCSRFToken, additionalHeaders, body) {
+    return new Promise((resolve, reject) => {
+      const options = {
+        method: method,
+        url: `${this.base}/${url.charAt(0) === '/' ? url.slice(1) : url}`,
+        auth: this.credentials,
+        headers: {
+          'X-CSRF-Token': XCSRFToken
```

Comment:
I think we can make `XCSRFToken` be part of the `additionalHeaders` object, since it's only required for unsafe methods (everything other than `GET`). We probably don't want to invoke `getXCSRFToken()` on every `GET`.